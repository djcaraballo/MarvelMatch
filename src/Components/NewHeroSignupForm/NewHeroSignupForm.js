import React, { Component } from 'react'

class NewHeroSignupForm extends Component {
  constructor(props) {
    super (props)
    this.state = {
      name: '',
      email: '',
      password: '',
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    //function that adds to store
    this.setState({
      name: '',
      email: '',
      password: '',
    })
  }

  render () {
    const { name, email, password } = this.state
    return (
      <form 
        className='new-hero-form'
        onSubmit={this.handleSubmit}
      >
        <input
          className='user-name-input'
          name='name'
          title='Enter first and last name'
          value={name}
          onChange={this.handleChange}
          placeholder='Enter first and last name'
        />
        <input
          className='user-email-input'
          name='email'
          title='Enter email address'
          value={email}
          onChange={this.handleChange}
          placeholder='Enter email address' />
        <input
          className='user-password-input'
          type='password'
          name='password'
          title='Enter password'
          value={password}
          onChange={this.handleChange}
          placeholder='Enter password' />
        <button>Match Me!</button>
      </form>
    )
  }
}

export default NewHeroSignupForm