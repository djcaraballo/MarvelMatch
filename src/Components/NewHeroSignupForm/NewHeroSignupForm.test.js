import React from 'react'
import { shallow } from 'enzyme'
import NewHeroSignupForm from './NewHeroSignupForm'

describe('NewHeroSignupForm Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NewHeroSignupForm />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the correct default state', () => {
    expect(wrapper.state().name).toEqual('')
    expect(wrapper.state().email).toEqual('')
    expect(wrapper.state().password).toEqual('')
  })

  describe('handleChange function', () => {
    xit('should set state with the correct names and values', () => {

    })
  })

  describe('handleSubmit function', () => {
    xit('should set state back to default', () => {

    })
  })
})