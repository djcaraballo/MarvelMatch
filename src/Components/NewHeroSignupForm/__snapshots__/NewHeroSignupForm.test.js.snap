// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewHeroSignupForm Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewHeroSignupForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="user-name-input"
          name="name"
          onChange={[Function]}
          placeholder="Enter first and last name"
          title="Enter first and last name"
          value=""
        />,
        <input
          className="user-email-input"
          name="email"
          onChange={[Function]}
          placeholder="Enter email address"
          title="Enter email address"
          value=""
        />,
        <input
          className="user-password-input"
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          title="Enter password"
          type="password"
          value=""
        />,
        <button>
          Match Me!
        </button>,
      ],
      "className": "new-hero-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "user-name-input",
          "name": "name",
          "onChange": [Function],
          "placeholder": "Enter first and last name",
          "title": "Enter first and last name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "user-email-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Enter email address",
          "title": "Enter email address",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "user-password-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Enter password",
          "title": "Enter password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Match Me!",
        },
        "ref": null,
        "rendered": "Match Me!",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="user-name-input"
            name="name"
            onChange={[Function]}
            placeholder="Enter first and last name"
            title="Enter first and last name"
            value=""
          />,
          <input
            className="user-email-input"
            name="email"
            onChange={[Function]}
            placeholder="Enter email address"
            title="Enter email address"
            value=""
          />,
          <input
            className="user-password-input"
            name="password"
            onChange={[Function]}
            placeholder="Enter password"
            title="Enter password"
            type="password"
            value=""
          />,
          <button>
            Match Me!
          </button>,
        ],
        "className": "new-hero-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "user-name-input",
            "name": "name",
            "onChange": [Function],
            "placeholder": "Enter first and last name",
            "title": "Enter first and last name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "user-email-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter email address",
            "title": "Enter email address",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "user-password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Enter password",
            "title": "Enter password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Match Me!",
          },
          "ref": null,
          "rendered": "Match Me!",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
